packages/react-tooltip/src/components/TooltipProvider/useTooltipManager.ts

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { ShowTooltipArgs, TooltipManager, TooltipTriggerReason } from '@fluentui/react-tooltip-trigger';
+
+const useTimeout = () => {
+  type TimeoutState = {
+    id: number | undefined;
+    set: (fn: () => void, delay: number) => void;
+    clear: () => void;
+  };
+
+  const state = React.useRef<TimeoutState>({
+    id: undefined,
+    set: (fn: () => void, delay: number) => {
+      state.clear();
+      state.id = window.setTimeout(fn, delay);
+    },
+    clear: () => {
+      if (state.id !== undefined) {
+        window.clearTimeout(state.id);
+        state.id = undefined;
+      }
+    },
+  }).current;
+
+  // Clean up the timeout when the component is unloaded
+  React.useEffect(() => state.clear, [state.clear]);
```

Comment:
I think this is what you wanted ? the current effect will just return a function handle

```suggestion
  React.useEffect(state.clear, [state.clear]);
```

Reply:
Yes this is what I wanted. The effect returns the function to be run when the component is unloaded, and I wanted clear to run when unloaded. However, I see how it can be confusing; I'll rewrite it to `return () => state.clear()` to be a little clearer.

Reply:
ah no worries, I think Ling.exe stopped working when I saw this, feel free to discard